kpi/views/v2/asset_permission_assignment.py

```diff
@@ -178,7 +178,11 @@ def bulk_assignments(self, request, *args, **kwargs):
         # one assignment fails.
         with transaction.atomic():
 
-            # First delete all assignments before assigning new ones.
+            # First, delete deployment backend flags if any
+            if self.asset.has_deployment:
+                self.asset.deployment.remove_from_flag()
```

Comment:
Wouldn't `self.asset.permissions.exclude(user__username=self.asset.owner.username)` below include `PERM_FROM_KC_ONLY`?

Reply:
Yes, it would, but it would also make `remove_perm()` fail. We have to trash `from_kc_only` separately beforehand